client/state/plugins/wporg/actions.js

```diff
@@ -52,3 +65,143 @@ export function fetchPluginData( pluginSlug ) {
 		}
 	};
 }
+
+/**
+ * Helper thunk for receiving a specific data or retrieve error of plugins list.
+ * Handles plugin list normalization internally.
+ *
+ * @param {string} category   Plugin category
+ * @param {number} page       Page (1-based)
+ * @param {string} searchTerm Search term
+ * @param {Array}  data       List of found plugins, not defined if there was an error
+ * @param {object} error      Error object, undefined if the plugins were fetched successfully
+ * @param {object} pagination Paginatioin data, as retrieved from the API response.
+ * @returns {object}          Action object
+ */
+function receivePluginsList( category, page, searchTerm, data, error, pagination = null ) {
+	return {
+		type: PLUGINS_WPORG_LIST_RECEIVE,
+		category,
+		page,
+		searchTerm,
+		data: normalizePluginsList( data ),
+		pagination,
+		error,
+	};
+}
+
+/**
+ * Retrieve a list of pliugins, identified by category and page number, or a search term.
+ *
+ * WP.org plugins can be filtered either by category or search term.
+ * Pagination is supported only for category queries.
+ * Category can be one of "featured", "popular", "new", "beta" or "recommended".
+ * Search term is an open text field.
+ *
+ * @param {string} category   Plugin category
+ * @param {number} page       Page (1-based)
+ * @param {string} searchTerm Search term
+ * @returns {Function} Action thunk
+ */
+export function fetchPluginsList( category, page, searchTerm ) {
+	return ( dispatch, getState ) => {
+		// Bail if we are currently fetching this plugins list
+		if ( isFetchingPluginsList( getState(), category, searchTerm ) ) {
+			return;
+		}
+
+		dispatch( {
+			type: PLUGINS_WPORG_LIST_REQUEST,
+			category,
+			page,
+			searchTerm,
+		} );
+
+		// @TODO: Remove when this flux action is completely reduxified
+		Dispatcher.handleViewAction( {
+			type: 'FETCH_WPORG_PLUGINS_LIST',
+			action: 'FETCH_WPORG_PLUGINS_LIST',
+			page,
+			category,
+			searchTerm,
+		} );
+
+		// The "Featured" category is managed by WP.com instead of WP.org
+		if ( 'featured' === category ) {
+			wpcom
```

Comment:
Nit: should we try to stick to use `async await` syntax to avoid deeper levels of nesting? No strong opinion on my side.

Reply:
I actually wanted to go that way, but then reconsidered when I found that I'd also have to refactor `fetchPluginsList`. Don't want to make this PR too large 😉 

It would be a nice improvement for the future, but definitely not a blocker IMHO. Also, I don't feel strongly either way, the current implementation I have suggested looks pretty clear to me.